oidc_provider/models.py

```diff
@@ -240,28 +248,79 @@ def at_hash(self):
 class UserConsent(BaseCodeTokenModel):
 
     user = models.ForeignKey(
-        settings.AUTH_USER_MODEL, verbose_name=_(u'User'), on_delete=models.CASCADE)
+        django_settings.AUTH_USER_MODEL, verbose_name=_(u'User'), on_delete=models.CASCADE)
     date_given = models.DateTimeField(verbose_name=_(u'Date Given'))
 
     class Meta:
         unique_together = ('user', 'client')
 
 
 class RSAKey(models.Model):
-
-    key = models.TextField(
-        verbose_name=_(u'Key'), help_text=_(u'Paste your private RSA Key here.'))
-
     class Meta:
+        abstract = True
         verbose_name = _(u'RSA Key')
         verbose_name_plural = _(u'RSA Keys')
 
+    @staticmethod
+    def model():
+        if settings.get("OIDC_RSA_CERT_STORE") == "database":
+            return RSAKeyDatabase
+        elif settings.get("OIDC_RSA_CERT_STORE") == "filesystem":
+            return RSAKeyFilesystem
+        raise Exception("unreachable")
```

Comment:
@zacps Just clarifying, this isn't required by an `abstract = True` model right? This is purely here to let other parts of the code pull the appropriate Model class reference back without checking the setting everywhere?

Reply:
Yep. I originally intended to implement this by overriding `__new__` to require any code changes elsewhere. Unfortunately there's quite a bit of nuance when overriding both `__init__` and `__new__` anywhere in the object sub-tree because you can't call `super(object, self).__new__(any-args)`. So if django's ORM overrode `__new__` in the future it wouldn't be called.

Reply:
Cool, I think this works well.

I wonder if a alternative name might make it more clear what was happening. `def implementation():` might be strictly accurate, but I am not sure it actually makes it clearer.

No strong feelings about a change really. Did you have any thoughts?